avocado/utils/configure_network.py

```diff
@@ -145,26 +175,55 @@ def is_link_up(self):
             return True
         return False
 
-    def interface_up(self):
-        cmd = "ifup %s" % self._interface
+    def get_hwaddr(self):
         try:
-            process.system(cmd, ignore_status=False, sudo=True)
+            interface_file = open(
+                '/sys/class/net/%s/address' % self._interface)
+            hwaddr = interface_file.read().strip()
+            interface_file.close()
+            return hwaddr
+        except OSError as ex:
+            raise NWException("interface not found : %s" % ex)
+
+    def set_hwaddr(self, hwaddr):
+        """
+        Utility which set Hw address to Interface
+
+        :param hwaddr: Pass Hardwae address for defined interface
+        """
+        try:
+            process.system('ifconfig %s hw ether %s' %
+                           (self._interface, hwaddr))
+        except Exception as ex:
+            raise NWException("ifdown fails: %s" % ex)
+
+    def add_hardware_address(self, maddr):
```

Comment:
some times you are using "hwaddr" and sometimes "hardware_address", we need to stick with one standard.